modules/gwlb/variables.tf

```diff
@@ -1,69 +1,148 @@
+  }
+  validation { # interval_in_seconds
+    condition = (var.health_probe.interval_in_seconds != null ?
+    var.health_probe.interval_in_seconds >= 5 && var.health_probe.interval_in_seconds <= 3600 : true)
+    error_message = "The `interval_in_seconds` property has to be between 5 and 3600 seconds (1 hour)."
+  }
+  validation { # probe_threshold
+    condition = (var.health_probe.probe_threshold != null ?
+    var.health_probe.probe_threshold >= 1 && var.health_probe.probe_threshold <= 100 : true)
+    error_message = "The `probe_threshold` property has to be between 1 and 100."
+  }
+  validation { # request_path
+    condition     = var.health_probe.protocol != "Tcp" ? var.health_probe.request_path != null : true
+    error_message = "The `request_path` property is required when protocol is set to \"Http\" or \"Https\"."
+  }
 }
 
 variable "backends" {
   description = <<-EOF
-  Map with backend configurations for the gateway load balancer. Azure GWLB rule can have up to two backends.
+  Map with backend configurations for the Gateway Load Balancer. Azure GWLB rule can have up to two backends.
+
   Following settings are available:
-  - `name`              - (Optional|string) Name of the backend. If not specified name is generated from `name` variable and backend key.
-  - `tunnel_interfaces` - (Required|map) Map with tunnel interfaces specs.)
-
-  Each tunnel interface specification consists of following settings (refer to [provider documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/lb_backend_address_pool#tunnel_interface) for details):
-  - `identifier` - (Required|int) Interface identifier.
-  - `port`       - (Required|int) Interface port.
-  - `type`       - (Required|string) Either "External" or "Internal".
-  
+  - `name`              - (`string`, required) name of the backend.
+                          If not specified name is generated from `name` variable and backend key.
+  - `tunnel_interfaces` - (`map`, required) map with tunnel interfaces.
+
+  Each tunnel interface specification consists of following settings:
+  - `identifier` - (`number`, required) interface identifier.
+  - `port`       - (`number`, required) interface port.
+  - `type`       - (`string`, required) either "External" or "Internal".
+
   If one backend is specified, it has to have both external and internal tunnel interfaces specified.
   For two backends, each has to have exactly one.
 
   On GWLB inspection enabled VM-Series instance, `identifier` and `port` default to:
   - `800`/`2000` for `Internal` tunnel type
   - `801`/`2001` for `External` tunnel type
-  Variable default reflects this configuration on GWLB side. Additionally, for VM-Series tunnel interface protocol is always VXLAN.
+
+  Variable default reflects this configuration on GWLB side.
+  Additionally, for VM-Series tunnel interface protocol is always VXLAN.
   EOF
   default = {
```

Comment:
should we have a default value here? shouldn't it be moved to examples? In other words, is this default value specific to this module? or maybe to the examples?

Reply:
I added default value for variable `backends` to have it as it's described in https://docs.paloaltonetworks.com/vm-series/10-2/vm-series-deployment/set-up-the-vm-series-firewall-on-azure/deploy-the-vm-series-firewall-with-the-azure-gwlb, but if you don't agree with that approach, I can remove it - can we discuss it ?